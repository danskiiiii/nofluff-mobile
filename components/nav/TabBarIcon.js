import Colors from '../../config/Colors';
import { Icon } from 'expo';
import React from 'react';

const TabBarIcon = props => (
  <Icon.Ionicons
    name={props.name}
    size={26}
    style={{ marginBottom: -3 }}
    color={props.focused ? Colors.tabSelected : Colors.tabDefault}
  />
);

export default TabBarIcon;
