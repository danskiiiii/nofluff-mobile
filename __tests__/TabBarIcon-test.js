import 'react-native';
import 'jsdom-global/register';

import { mount, shallow } from 'enzyme';

import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import React from 'react';
import TabBarIcon from '../components/nav/TabBarIcon';
import renderer from 'react-test-renderer';

describe('TabBarIcon snapshot', () => {
  jest.useFakeTimers();
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it('renders the TabBarIcon', async () => {
    const tree = renderer.create(<TabBarIcon />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

function getComponent(focused, name) {
  return <TabBarIcon focused={focused} name={name} />;
}

describe('TabBarIcon component', () => {
  it('renders', () => {
    const wrapper = shallow(getComponent(true, 'md-options'));

    expect(wrapper.exists()).toBe(true);
  });

  it('renders Icon', () => {
    const wrapper = mount(getComponent(true, 'md-options'));

    expect(wrapper.find('Icon').exists()).toBe(true);
  });
});
