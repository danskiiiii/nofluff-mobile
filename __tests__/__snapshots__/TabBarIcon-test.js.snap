// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarIcon snapshot renders the TabBarIcon 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
/>
`;
